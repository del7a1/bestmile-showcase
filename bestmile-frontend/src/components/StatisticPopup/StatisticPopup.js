import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Popup from 'react-popup'

export default class StatisticPopup extends Component {

  render() {
    return (
      <div>
        <label for="missionsCounter">Missions completed</label>
        <input id="missionsCounter" type="text" className="mm-popup__input" value={this.props.missionsCounter}/>

        <label for="totalPassengerCount">Total number of passengers</label>
        <input id ="totalPassengerCount" type="text" className="mm-popup__input" value={this.props.totalPassengerCount}/>

        <label for="totalDistance">Total traveled distance (miles)</label>
        <input id="totalDistance" type="text" className="mm-popup__input" value={this.props.totalDistance}/>
      </div>
    );
  }

}

StatisticPopup.propTypes = {
  missionsCounter: PropTypes.number,
  totalPassengerCount: PropTypes.number,
  totalDistance: PropTypes.number
};

Popup.registerPlugin('statistics', (missionsCounter, totalPassengerCount, totalDistance) => {

  Popup.create({
    title: 'Statistics',
    content: <StatisticPopup missionsCounter={missionsCounter} totalPassengerCount={totalPassengerCount} totalDistance={totalDistance} />
  });

});

