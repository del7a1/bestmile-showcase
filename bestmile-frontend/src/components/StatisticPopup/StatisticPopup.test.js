import React from 'react';
import { shallow } from 'enzyme';
import StatisticPopup from './StatisticPopup.js';
import Popup from 'react-popup'

describe('StatisticPopup', () => {

 it('Should be defined', () => {
   expect(StatisticPopup).toBeDefined();
 });

 it('Should render correctly', () => {
   const wrapper = shallow(
     <StatisticPopup missionsCounter={1} totalPassengerCount={2} totalDistance={3} />
   );
   expect(wrapper).toMatchSnapshot();
 });

 it('Should contains input fields with statistics', () => {
   const wrapper = shallow(
     <StatisticPopup missionsCounter={1} totalPassengerCount={2} totalDistance={3} />
   );
   expect(wrapper.find('#missionsCounter').props().value).toEqual(1);
   expect(wrapper.find('#totalPassengerCount').props().value).toEqual(2);
   expect(wrapper.find('#totalDistance').props().value).toEqual(3);
 });

});
