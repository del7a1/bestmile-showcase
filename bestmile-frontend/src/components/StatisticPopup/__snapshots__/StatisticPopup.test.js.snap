// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticPopup Should render correctly 1`] = `
<div>
  <label
    for="missionsCounter"
  >
    Missions completed
  </label>
  <input
    className="mm-popup__input"
    id="missionsCounter"
    type="text"
    value={1}
  />
  <label
    for="totalPassengerCount"
  >
    Total number of passengers
  </label>
  <input
    className="mm-popup__input"
    id="totalPassengerCount"
    type="text"
    value={2}
  />
  <label
    for="totalDistance"
  >
    Total traveled distance (miles)
  </label>
  <input
    className="mm-popup__input"
    id="totalDistance"
    type="text"
    value={3}
  />
</div>
`;
